File: test-projects/FSharpTest/StatisticsTest.fs
module StatisticsTest

open Xunit
open FsUnit.Xunit
open Synapses
open FSharp.Data
open FSharpx.Collections

type ``statistics tests``() =
    
    let expectedWithOutputValues =
            Seq.ofList [ ( [ 0.0; 0.0; 1.0], [ 0.0; 0.0; 1.0] )
                         ( [ 0.0; 0.0; 1.0], [ 0.0; 1.0; 1.0] ) ]
    
    [<Fact>]
    let ``root mean square error``() =
        Statistics.rootMeanSquareError expectedWithOutputValues
        |> should equal 0.7071067811865476

File: test-projects/remote-deps/CSharpTest/test/StatisticsTest.cs
using System.Collections.Generic;
using SynapsesCSharp;
using Xunit;

namespace CSharpTest.test
{
    public class StatisticsTest
    {
        static IEnumerable<(double[], double[])> expectedWithOutputValues =
            new List<(double[], double[])>()
            {
                (new double[] {0.0, 0.0, 1.0}, new double[] {0.0, 0.0, 1.0}),
                (new double[] {0.0, 0.0, 1.0}, new double[] {0.0, 1.0, 1.0})
            };

        [Fact]
        public void rootMeanSquareError()
        {
            Assert.Equal(
                0.7071067811865476,
                Statistics.rootMeanSquareError(expectedWithOutputValues)
            );
        }
    }
}